src/plugins/run/annotations.ts

```diff
@@ -0,0 +1,203 @@
+import { Dictionary } from '@stoplight/types';
+import { parse } from '@stoplight/yaml';
+import * as unified from 'unified';
+
+import { MDAST } from '../../ast-types';
+
+// When we have bandwidth, might make sense to look into the new lower level pattern made possible by micromark ecosystem
+// Example in the gfm plugin - https://github.com/remarkjs/remark-gfm/blob/main/index.js#L30-L32
+export const smdAnnotations: unified.Attacher = function () {
+  return function transform($root) {
+    const root = $root as MDAST.Root;
+    const nodes = root.children;
+
+    const processed: MDAST.Content[] = [];
+
+    let inTab: boolean = false;
+    let skipNext: boolean = false;
+
+    // temporary variable for storing current tabContainer
+    let tabPlaceholder: MDAST.Tabs = {
+      type: 'tabs',
+      data: {
+        hName: 'tabs',
+      },
+      children: [
+        {
+          type: 'tab',
+          data: {
+            hName: 'tab',
+          },
+          children: [],
+        },
+      ],
+    };
+
+    for (const i in nodes) {
+      if (!nodes[i]) continue;
```

Comment:
any reason we cannot use `for of`?
```suggestion
    for (const [i, node] of nodes.entries()) {
```